### PR TITLE
[Merged by Bors] - Move prepass functions to prepass_utils

### PR DESCRIPTION
**Pull request by IceSentry**
# Objective

- The functions added to utils.wgsl by the prepass assume that mesh_view_bindings are present, which isn't always the case
- Fixes https://github.com/bevyengine/bevy/issues/7353

## Solution

- Move these functions to their own `prepass_utils.wgsl` file

### CODE REVIEW

**Review by torsteingrindvik — APPROVED** [review]
From a quick look this looks good to me and looks like it will solve my linked issue.

**Review by JMS55 — APPROVED** [review]

File: crates/bevy_pbr/src/prepass/prepass_utils.wgsl [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+#ifndef DEPTH_PREPASS
+fn prepass_depth(frag_coord: vec4<f32>, sample_index: u32) -> f32 {
+#ifdef MULTISAMPLED
+    let depth_sample = textureLoad(depth_prepass_texture, vec2<i32>(frag_coord.xy), i32(sample_index));
+#else
+    let depth_sample = textureLoad(depth_prepass_texture, vec2<i32>(frag_coord.xy), 0);
+#endif
+    return depth_sample;
+}
+#endif // DEPTH_PREPASS
```
Comment (james7132) [inline crates/bevy_pbr/src/prepass/prepass_utils.wgsl:23]:
```suggestion
#endif // DEPTH_PREPASS

```

File: crates/bevy_pbr/src/render/utils.wgsl [hunk 2]
```diff
@@ -24,26 +24,4 @@ fn random1D(s: f32) -> f32 {
...
-#ifdef MULTISAMPLED
-    let depth_sample = textureLoad(depth_prepass_texture, vec2<i32>(frag_coord.xy), i32(sample_index));
-#else
-    let depth_sample = textureLoad(depth_prepass_texture, vec2<i32>(frag_coord.xy), 0);
-#endif
-    return depth_sample;
-}
-#endif // DEPTH_PREPASS
\ No newline at end of file
+}
```
Comment (james7132) [inline crates/bevy_pbr/src/render/utils.wgsl:27]:
```suggestion
}

```

**Review by james7132 — APPROVED** [review]

**Review by james7132 — APPROVED** [review]
bors r+

**Pull request closed without merging**